src/common/database-service/DatabaseTimeSeriesActions.ts

```diff
@@ -110,6 +121,11 @@ export class DatabaseTimeSeriesActions extends DatabaseActions {
      * then compute the result set + any M4-like reduction on it.
      * We first create a resultset of zero-values,
      * then join this result set against the empirical counts.
+     *
+     * Limitation: due to the use of `date_trunc()` in the `series` CTE,
```

Comment:
why is this?

Reply:
E.g. for a desired granularity of 10 minutes:
- date_trunc("minute", ...) will truncate every timestamp to something like 1:00pm, 1:01pm, 1:02pm...
- the `GROUP BY` will result in 60 different time buckets in an hour
- we currently pick out the 1:00pm, 1:10pm, 1:20pm buckets and we lose the data in 1:01pm-1:09pm (and similar) buckets